dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/ValidationUtils.java

```diff
@@ -437,11 +437,11 @@ else if ( ValueType.INTEGER_ZERO_OR_POSITIVE == valueType && !MathUtils.isZeroOr
         {
             return "value_not_zero_or_positive_integer";
         }
-        else if ( ValueType.BOOLEAN == valueType && !MathUtils.isBool( value ) )
+        else if ( ValueType.BOOLEAN == valueType && !MathUtils.isBool( value.toLowerCase() ) )
         {
             return "value_not_bool";
         }
-        else if ( ValueType.TRUE_ONLY == valueType && !DataValue.TRUE.equals( value ) )
+        else if ( ValueType.TRUE_ONLY == valueType && !DataValue.TRUE.equals( value.toLowerCase() ) )

```

Comment:
Have you verified that we are not introducing a null pointer vulnerability here?

Reply:
Hi @larshelge , actually at the beginning of this method we are already checking if the "value" is null.
But, I can make it safer for sure. I will fix it in a few minutes.

Thanks

Reply:
Fix added. Ready for another review.